Improve upon existing code as per review comments

ayush000 opened the pull request:
- Replace `for` with `doseq`
- Replace `if` block with `when` block
- Don't return anything in main function
- Read list of items to watch from a JSON file

Commit:
Implement suggested improvements

Commit:
Fetch shopping list from a JSON file

rubaljabbal commented on src/github_amazon_bot/core.clj:
```diff
@@ -26,13 +24,12 @@
+      (when ipad-is-available
+        (try
+          (api/send-text token chat-id (str name " is available! Please buy it from " url))
+          (catch Exception))))))
```
You need catch Exceptions in a var.

rubaljabbal commented on src/github_amazon_bot/core.clj:
```diff
@@ -16,20 +17,25 @@
         (try
           (api/send-text token chat-id (str name " is available! Please buy it from " url))
-          (catch Exception))))))
+          (catch Exception ex))))))
```
I see that you've fixed it already :)

ayush000 replied:
Linter was showing a warning for unused var. Not sure how to fix that, so leaving it as is.

rubaljabbal reviewed (approved):
LGTM

Commit:
Don't count characters in text-to-search

This is not needed. re-find returns nil if the value isn't present. This
can just be typecasted to boolean.
